src/offer/offerService.js

```diff
@@ -25,6 +27,43 @@ export class OfferService {
     }
   };
 
+  complete = async (id, userNameTrader) => {
+    if ((!id) || (!userNameTrader)) {
+      throw new InvalidDataError(INVALID_DATA, OFFERS);
+    }
+
+    const offer = this.offerRepository.getOffer(id);
+
+    const { offerItems, requestItems } = offer;
+
+    const exist = await this.userRepository.exist(userNameTrader);
+
+    if (!exist) {
```

Comment:
if (!existUserNameTrader) or if(!await this.userRepository.exist(userNameTrader))